### PR TITLE
UCT/UCP: atomic cleanup

### PR DESCRIPTION
**Pull request by hoopoepg**
- removed deprecated tl entries
- removed deprecated ATOMIC macro

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6906/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6907/ for details (Mellanox internal link).

File: src/tools/perf/libperf.c [hunk 1]
```diff
@@ -407,7 +409,12 @@ static ucs_status_t uct_perf_test_check_capabilities(ucx_perf_params_t *params,
         return status;
     }
 
-    if (!ucs_test_all_flags(attr.cap.flags, required_flags) || !required_flags) {
+    if (!ucs_test_all_flags(attr.cap.flags, required_flags)            ||
+        !ucs_test_all_flags(attr.cap.atomic32.op_flags, atomic_op32)   ||
+        !ucs_test_all_flags(attr.cap.atomic64.op_flags, atomic_op64)   ||
+        !ucs_test_all_flags(attr.cap.atomic32.fop_flags, atomic_fop32) ||
+        !ucs_test_all_flags(attr.cap.atomic64.fop_flags, atomic_fop64) ||
+        !(required_flags | atomic_op32 | atomic_op64 | atomic_fop32 | atomic_fop64)) {
```
Comment (yosefe) [inline src/tools/perf/libperf.c]: pls add details which flags are missing

**Comment by yosefe** [comment]
@shamisp can you pls look at the API change - removing the old atomics functions
@hjelmn - after this change, would need to update uct btl to use new atomics

**Comment by shamisp** [comment]
👍

File: src/tools/perf/libperf.c (hunk 1 above)
Comment (hoopoepg) [reply src/tools/perf/libperf.c]: @yosefe added diagnostic, error message will looks like
```UCX  ERROR rc/mlx5_0:1 does not support operation am short```

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6931/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6932/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6933/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6935/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6938/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
failed due to missing header

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6943/ for details (Mellanox internal link).

**Pull request merged**